rbc/src/main/java/com/library/rbc/service/BookService.java

```diff
@@ -34,4 +39,61 @@ public BookDto addNewBook(BookDto bookDTO) {
     bookRepository.save(savedBook);
     return bookDTO;
   }
+
+  public Page<BookDto> getBooksBy(Pageable pageable, List<String> bookCategories,
+      List<String> bookStatuses) {
+    return getBooksByCategoriesAndStatuses(pageable, bookCategories, bookStatuses);
+  }
+
+  private Page<BookDto> getBooksByCategoriesAndStatuses(Pageable pageable,
```

Comment:
Consider refactoring `getBooksByCategoriesAndStatuses` using the following ideas:
- Avoid nested if statements
- Extract methods if needed